blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/LocatorFetchRunnable.java

```diff
@@ -155,6 +141,14 @@ public void run() {
         if (log.isDebugEnabled())
             log.debug("Finished {} rollups for (gran,slot,shard) {} in {}", new Object[] {rollCount, parentSlotKey, System.currentTimeMillis() - waitStart});
 
+        finishExecution(waitStart, executionContext);
+    }
+
+    public void waitForRollups() throws InterruptedException {
+        Thread.currentThread().sleep(LOCATOR_WAIT_FOR_ALL_SECS * 1000);
+    }
```

Comment:
Someone could totally mess with us by calling this repeatedly. We shouldnt expose these kinda methods. If you want control, have LOCATOR_WAIT_FOR_ALL_SECS in a separate constructor. For tests you can set it to zero if you dont want it to wait.